backend/module/bot/slackbot/slackbot.go

```diff
@@ -0,0 +1,249 @@
+package slackbot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	slackbotv1 "github.com/lyft/clutch/backend/api/bot/slackbot/v1"
+	slackbotconfigv1 "github.com/lyft/clutch/backend/api/config/module/bot/slackbot/v1"
+	"github.com/lyft/clutch/backend/gateway/log"
+	"github.com/lyft/clutch/backend/module"
+	"github.com/lyft/clutch/backend/service"
+	"github.com/lyft/clutch/backend/service/bot"
+)
+
+const (
+	Name = "clutch.module.bot.slackbot"
+	// request headers used to verify a request came from Slack
+	hSignature = "x-Slack-Signature"
+	hTimestamp = "x-Slack-Request-Timestamp"
+	// slack event type for DMs with the bot
+	dmMessage = "im"
+)
+
+func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (module.Module, error) {
+	config := &slackbotconfigv1.Config{}
+	if cfg != nil {
```

Comment:
is a nil config valid?

Reply:
ah nope! 
will update, thx for catching